packages/plugin-core/src/commands/MoveHeader.ts

```diff
@@ -0,0 +1,311 @@
+   * At each level _position_ is omitted as this can change if
+   * you are comparing from two different trees.
+   * @param a first {@link Node} to compare
+   * @param b second {@link Node} to compare
+   * @returns boolean
+   */
+  hasIdenticalChildren = (a: Node, b: Node): boolean => {
+    if (_.isEqual(Object.keys(a).sort(), Object.keys(b).sort())) {
+      const aOmit = _.omit(a, ["position", "children"]);
+      const bOmit = _.omit(b, ["position", "children"]);
+      if (_.isEqual(aOmit, bOmit)) {
+        if (_.has(a, "children")) {
+          return _.every(a.children as Node[], (aChild, aIndex) => {
+            const bChild = (b.children as Node[])[aIndex];
+            return this.hasIdenticalChildren(aChild, bChild);
+          });
+        }
+        return true;
+      } else {
+        return false;
+      }
+    } else {
+      return false;
+    }
+  };
+
+  async gatherInputs(opts: CommandInput): Promise<CommandOpts | undefined> {
+    const engine = getEngine();
+    const { editor, selection } = VSCodeUtils.getSelection();
+    if (!editor) throw this.noActiveNoteError;
+    if (!selection) throw this.headerNotSelectedError;
+
+    // const text = editor.document.getText();
+    const line = editor.document.lineAt(selection.start.line).text;
+    const maybeNote = VSCodeUtils.getNoteFromDocument(editor.document);
+    if (!maybeNote) {
+      throw this.noActiveNoteError;
+    }
+
+    const proc = this.getProc(engine, maybeNote);
+    const parsedLine = proc.parse(line);
+    let targetHeader: Heading | undefined;
+    visit(parsedLine, [DendronASTTypes.HEADING], (heading: Heading) => {
+      targetHeader = heading;
+      return false;
+    });
+    if (!targetHeader) {
+      throw this.headerNotSelectedError;
+    }
+    const tree = proc.parse(maybeNote.body);
+    let headerFound = false;
+    let foundHeaderIndex: number | undefined;
+    let nextHeaderIndex: number | undefined;
+    visit(tree, (node, index) => {
+      if (nextHeaderIndex) {
+        return;
+      }
+      const depth = node.depth as Heading["depth"];
+      if (!headerFound) {
+        if (node.type === DendronASTTypes.HEADING) {
+          if (
+            depth === targetHeader!.depth &&
+            this.hasIdenticalChildren(node, targetHeader!)
+          ) {
+            headerFound = true;
+            foundHeaderIndex = index;
+            return;
+          }
+        }
+      } else if (node.type === DendronASTTypes.HEADING) {
+        if (foundHeaderIndex) {
+          if (depth <= targetHeader!.depth) nextHeaderIndex = index;
+        }
+      }
+    });
+
+    const nodesToMove = nextHeaderIndex
+      ? (tree.children as any[]).splice(
+          foundHeaderIndex!,
+          nextHeaderIndex! - foundHeaderIndex!
+        )
+      : (tree.children as any[]).splice(foundHeaderIndex!);
+
+    const lookupController = LookupControllerV3.create({
+      nodeType: "note",
+      disableVaultSelection: true,
```

Comment:
Do we actually want this disabled? Can we move the header to a note in a different vault?

Reply:
+1 on this. that should be possible

Reply:
Blocked out a short session to see if this is trivial but it doesn't seem like it. Will be adding this in my backlog as a next item.